File: api/load_result.py
# api/load_result.py
import os
import json

RESULT_DIR = "results"

def handler(request):
    try:
        if not os.path.exists(RESULT_DIR):
            return {
                "statusCode": 200,
                "body": json.dumps([])
            }

        files = [f for f in os.listdir(RESULT_DIR) if f.endswith(".json")]
        result_list = []

        for fname in sorted(files, reverse=True):
            with open(os.path.join(RESULT_DIR, fname), "r", encoding="utf-8") as f:
                data = json.load(f)
                result_list.append(data)

        return {
            "statusCode": 200,
            "headers": {"Content-Type": "application/json"},
            "body": json.dumps(result_list)
        }

    except Exception as e:
        return {
            "statusCode": 500,
            "body": f"에러 발생: {str(e)}"
        }

File: api/load_result.js
import fs from "fs";
import path from "path";

export default function handler(req, res) {
  try {
    const resultDir = path.join(process.cwd(), "results");
    if (!fs.existsSync(resultDir)) {
      return res.status(200).json([]);
    }

    const files = fs.readdirSync(resultDir).filter(f => f.endsWith(".json"));
    const resultList = files.sort().reverse().map(fname => {
      const content = fs.readFileSync(path.join(resultDir, fname), "utf-8");
      return JSON.parse(content);
    });

    res.status(200).json(resultList);
  } catch (err) {
    res.status(500).send(`에러 발생: ${err.message}`);
  }
}
